lib/routes/account.js

```diff
@@ -406,177 +398,48 @@ module.exports = (log, db, mailer, Password, config, customs, checkPassword, pus
-                  throw error.cannotLoginWithSecondaryEmail()
-                }
-
-                accountRecord = result
-              },
-              function (err) {
-                if (err.errno === error.ERRNO.ACCOUNT_UNKNOWN) {
-                  customs.flag(request.app.clientAddress, {
-                    email: email,
-                    errno: err.errno
-                  })
-
-                  // We rate-limit attempts to check whether an account exists, so
-                  // we have to be careful to re-throw any customs-server errors here.
-                  // We also have to be careful not to leak info about whether the account
-                  // existed, for example by saying sign-in unblock is unavailable on
-                  // accounts that don't exist. We pass a fake uid into the feature-flag
-                  // test to mask whether the account existed.
-                  if (customsErr) {
-                    throw customsErr
-                  }
-
-                  throw err
-                }
-                throw err
-              }
-            )
+        function checkCustomsAndLoadAccount() {
+          return signinUtils.checkCustomsAndLoadAccount(request, email).then((res) => {
+            accountRecord = res.accountRecord
+            // Remember whether they did a signin-unblock,
+            // because we can use it to bypass token verification.
+            didSigninUnblock = res.didSigninUnblock
+          })
         }
 
-        function checkUnblockCode() {
-          if (unblockCode) {
-            return db.consumeUnblockCode(accountRecord.uid, unblockCode)
-              .then(
-                (code) => {
-                  if (requestNow - code.createdAt > unblockCodeLifetime) {
-                    log.info({
-                      op: 'Account.login.unblockCode.expired',
-                      uid: accountRecord.uid
-                    })
-                    throw error.invalidUnblockCode()
-                  }
-                  didSigninUnblock = true
-                  return request.emitMetricsEvent('account.login.confirmedUnblockCode')
-                }
-              )
-              .catch(
-                (err) => {
-                  if (err.errno === error.ERRNO.INVALID_UNBLOCK_CODE) {
-                    return request.emitMetricsEvent('account.login.invalidUnblockCode')
-                      .then(() => {
-                        customs.flag(request.app.clientAddress, {
-                          email: email,
-                          errno: err.errno
-                        })
-                        throw err
-                      })
-                  }
-                  throw err
-                }
+        function checkEmailAndPassword() {
+          return signinUtils.checkEmailAddress(accountRecord, email, originalLoginEmail)
+            .then(() => {
+              password = new Password(
```

Comment:
Previously this route was calling `new Password` twice - once when invoking `checkPassword` to check the password, and again when creating the keyFetchToken.  I've pulled it out into a local variable here, so that we save ourselves a second invocation of the expensive scrypt operation that's hiding behind this object.  It would be interesting to see if we can notice a decrease in response times with this in prod.